client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -935,6 +935,33 @@ Undocumented.prototype.setPrimaryDomain = function( siteId, domain, fn ) {
 	this.wpcom.req.post( '/sites/' + siteId + '/domains/primary', {}, { domain: domain }, fn );
 };
 
+/**
+ * Fetch preview markup for a site
+ *
+ * @param {int} siteId The site ID
+ * @param {string} slug Optional. The theme slug to preview
+ * @param {object} postData Optional. The customization data to send
+ * @return {Promise} A Promise to resolve when complete
+ */
+Undocumented.prototype.fetchPreviewMarkup = function( siteId, slug, postData ) {
+	debug( '/sites/:site_id/previews/mine' );
+	return new Promise( ( resolve, reject ) => {
+		const endpoint = `/sites/${siteId}/previews/mine`;
+		const query = { path: slug };
+		const isPreviewCustomized = ( postData && Object.keys( postData ).length > 0 );
+		const { post, get } = this.wpcom.req;
```

Comment:
Oh yeah, interesting indeed. I didn't expect that `get` or `post` would use `this`, but [there it is](https://github.com/Automattic/wpcom.js/blob/master/lib/util/request.js#L29). It looks like maybe binding to the right-hand-side object is part of object destructuring assignment.